api/krusty/helmchartinflationgenerator_test.go

```diff
@@ -416,6 +416,123 @@ metadata:
 `)
 }
 
+// Reference: https://github.com/kubernetes-sigs/kustomize/issues/5163
+func TestHelmChartInflationGeneratorForMultipleChartsDifferentVersion(t *testing.T) {
```

Comment:
Added test case based on @koba1t [suggestion](https://github.com/kubernetes-sigs/kustomize/pull/5293#issuecomment-1780990501) here to cover multiple charts for different versions, as well as reference chart using version but without repo.